src/app/profile/payment-methods/bank-account/bank-account.component.js

```diff
@@ -0,0 +1,73 @@
+import angular from 'angular';
+import template from './bank-account.tpl';
+import displayAddressComponent from 'common/components/display-address/display-address.component';
+import editPaymentMethodModal from 'common/components/paymentMethods/editPaymentMethod/editPaymentMethod.modal.component';
+import deletePaymentMethodModal from 'common/components/paymentMethods/deletePaymentMethod/deletePaymentMethod.modal.component.js';
+import giveModalWindowTemplate from 'common/templates/giveModalWindow.tpl';
+import recurringGiftsComponent from '../recurring-gifts/recurring-gifts.component';
+
+
+class BankAccountController{
+
+  /* @ngInject */
+  constructor(envService, $uibModal){
+    this.isCollapsed = true;
+    this.$uibModal = $uibModal;
+    this.imgDomain = envService.read('imgDomain');
+    this.submissionError = {error: ''};
+  }
+
+  getExpiration(){
+    return `${this.model['expiry-month']}/${this.model['expiry-year']}`;
+  }
+
+  editPaymentMethod() {
+    this.editPaymentMethodModal = this.$uibModal.open({
+      component: 'editPaymentMethodModal',
+      windowTemplateUrl: giveModalWindowTemplate.name,
+      size: 'lg',
+      resolve: {
+        model: () => this.model,
+        paymentType: () => 'bankAccount',
+        submissionError: () => this.submissionError
+      }
+    });
+  }
+
+  deletePaymentMethod(){
+    this.deletePaymentMethodModal = this.$uibModal.open({
```

Comment:
@OzzieOrca , I made a hardcoded solution for now but I think we can combine error handling and success messages into one component. BTW, according to the David Sudarma we will fade out success messages after 60 seconds.